pkg/util/helpers.go

```diff
@@ -171,3 +176,41 @@ func AddCertificateErrorHandler(fatal bool) {
 		// The generic handler has already logged the error, no need to repeat if we don't want extra detail
 	})
 }
+
+// StartMetricsAndProfileServer starts an HTTP server providing access to Prometheus metrics
+// and/or profiling information.
+// The returned server should be shut down on process exit, using StopMetricsAndProfileServer().
+func StartMetricsAndProfileServer(metrics, profile bool, port int) *http.Server {
```

Comment:
Users don't actually need the `http.Server ` other than to close it so how about returning a function that closes it instead. This obviates the need for a separate Stop* function.

```suggestion
type StopFunc func()

func StartHTTPServer(metrics, profile bool, port int) StopFunc{
     ...
    return func() {
        if err := srv.Shutdown(context.TODO()); err != nil {
		logger.Errorf(err, "Error shutting down metrics/profile HTTP server")
	}
    }
}
```

Reply:
We could put this in an `http` package:` http.StartServer`. I've thought about moving the functions in the `util` package to other, suitably-named packages.